lib/src/network/backends.rs

```diff
@@ -125,21 +125,34 @@ impl BackendMap {
       return self.backend_from_app_id(app_id);
     }
   }
+
+  pub fn set_load_balancing_policy_for_app(&mut self, app_id: &str, lb_algo: LoadBalancingAlgorithms) {
+    // The application can be create before the backends were registered because of the async config messages.
+    // So when we set the load balancing policy, we have to create the backend list if if it doesn't exist yet.
+    let app_backends = self.get_or_create_backend_list_for_app(app_id);
+    app_backends.set_load_balancing_policy(lb_algo);
+  }
+
+  pub fn get_or_create_backend_list_for_app(&mut self, app_id: &str) -> &mut BackendList {
+    self.backends.entry(app_id.to_string()).or_insert(BackendList::new())
+  }
 }
 
 const MAX_FAILURES_PER_BACKEND: usize = 10;
 
 #[derive(Debug)]
 pub struct BackendList {
-  pub backends:  Vec<Rc<RefCell<Backend>>>,
-  pub next_id:   u32,
+  pub backends:       Vec<Rc<RefCell<Backend>>>,
+  pub next_id:        u32,
+  pub load_balancing: Box<LoadBalacingAlg>,
 }
 
 impl BackendList {
   pub fn new() -> BackendList {
     BackendList {
-      backends:  Vec::new(),
-      next_id:   0,
+      backends:       Vec::new(),
+      next_id:        0,
+      load_balancing: Box::new(RandAlg{}),
```

Comment:
maybe there's a better name than "RandAlg"

Reply:
Can you make a proposal ?

Reply:
the first thing would be to write complete words :)